### PR TITLE
Add tests for role searches and support for .rb role files

### PR DESCRIPTION
**Pull request by ghost**
I'm open to suggestions on how to do the .rb role files better, assuming the filenames are the same as the rolenames isn't necessarily correct.

### CODE REVIEW

**Comment by ghost** [comment]
That Travis error doesn't look like it's anything that I've touched in this change.  It's claiming "Gem::InstallError: mime-types requires Ruby version >= 1.9.2." but it's done two builds on Ruby 1.8 and one of them works just fine.

**Comment by tobami** [comment]
I just triggered a rebuild on travis and everything is green now

File: tests/test_roles.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
+#
+# Copyright 2011, edelight GmbH
+#
+# Authors:
+#       Markus Korn <markus.korn@edelight.de>
```
Comment (tobami) [inline tests/test_roles.rb:5]: Backdated copyright notice? :smile:

**Comment by tobami** [comment]
Otherwise looking good, though you could update the roles section in the README.md to make clear that .rb roles are supported

**Comment by tobami** [comment]
@kaerast do you want to add a note to the README yet? Are you still interested in integrating this feature?

File: tests/test_roles.rb (hunk 1 above)
Comment (ghost) [inline tests/test_roles.rb:5]: That's just a copy/paste from the copyright notices elsewhere in the project.  I didn't want to pollute the project with my own copyright.

**Comment by ghost** [comment]
Sorry it's taken so long for me to get round to updating the README, I've simply not had chance to come back to this.

**Comment by tobami** [comment]
No problem, thanks for it. It conflicts because I move yesterday "Nodes" below "Databags" I guess. Can you try to resolve it by merging master into your branch? Otherwise I can just merge into master and resolve conflicts at that point. It's just cleaner the other way

**Comment by michaelglass** [comment]
this looks great.  merging.

**Pull request merged**